### PR TITLE
feat: [SRM-13419]: Log fields templates

### PR DESCRIPTION
**Pull request by praneshg239**
#### Screenshots

NA

https://user-images.githubusercontent.com/95267551/209221524-45974a8d-1775-4468-b5fa-33b3f9bb1410.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSourceContext.tsx [hunk 1]
```diff
@@ -5,17 +5,7 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
-import { createContext } from 'react'
-import type { FormikHelpers } from 'formik'
-import type { CommonHealthSourceConfigurations } from './CommonHealthSource.types'
-
 export enum CommonHealthSourceContextFields {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSourceContext.tsx:8]: we should rename this file as this doesn't have any context related change

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonHealthSourceContext/CommonHealthSourceContext.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import type { FormikHelpers } from 'formik'
+import React, { useCallback, useState } from 'react'
+import type { CommonHealthSourceConfigurations } from '@cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.types'
+
+export interface CommonHealthSourceContextType {
+  isQueryRuntimeOrExpression?: boolean
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonHealthSourceContext/CommonHealthSourceContext.tsx:6]: we should add isConnectorRuntimeOrExpression as well

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/LogsTable/components/JsonSelectorWithDrawer.tsx [hunk 3]
```diff
@@ -29,16 +42,80 @@ interface JsonSelectorWithDrawerProps {
...
 
   const { fieldMappings, disableFields, jsonData } = props
 
+  const filteredFieldsMapping = fieldMappings?.filter(field => field.type === FIELD_ENUM.JSON_SELECTOR)
+
   const isDisabled = disableFields
 
-  const filteredFieldsMapping = fieldMappings?.filter(field => field.type === FIELD_ENUM.JSON_SELECTOR)
+  const [multiTypeRecord, setMultiTypeRecord] = useState<LogFieldsMultiTypeState | null>(
+    (): LogFieldsMultiTypeState | null => {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/LogsTable/components/JsonSelectorWithDrawer.tsx:60]: Move to function and use

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonHealthSourceContext/CommonHealthSourceContext.tsx (hunk 2 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonHealthSourceContext/CommonHealthSourceContext.tsx:6]: No need, as it is coming from sourceTab context

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSourceContext.tsx (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSourceContext.tsx:8]: This file is deleted and moved `CommonHealthSourceContextFields` to constants

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/LogsTable/components/JsonSelectorWithDrawer.tsx (hunk 3 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/LogsTable/components/JsonSelectorWithDrawer.tsx:60]: Done

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSourceContext.tsx (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSourceContext.tsx]: Done

**Review by deepesh-ui — DISMISSED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Review by karansaraswat19 — APPROVED** [review]

**Pull request merged**